spec/models/search/bookmark_search_form_spec.rb

```diff
@@ -13,6 +13,126 @@
 
   describe "bookmarkable_search_results" do
     describe "sorting" do
+      context "by word count" do
+        let(:tag) { create(:canonical_fandom) }
+
+        let!(:work5) { create(:work, fandom_string: tag.name, chapter_attributes: { content: "one two three four five" }) }
+        let!(:work10r) { create(:work, fandom_string: tag.name, restricted: true, chapter_attributes: { content: "one two three four five six seven eight nine ten" }) }
+        let!(:work10) { create(:work, fandom_string: tag.name, title: "Ten", chapter_attributes: { content: "one two three four five six seven eight nine ten" }) }
+        # bookmark of public work "Ten" has word_count 10
+        let!(:work_bookmark) { create(:bookmark, bookmarkable: work10) }
+
+        let!(:series) { create(:series, title: "Series to be bookmarked", works: [work5, work10r]) }
+        # bookmark of series "Series to be bookmarked" has word_count 5 or 15
+        # depending on whether work10r (a restricted work) is visible
+        let!(:series_bookmark) { create(:bookmark, bookmarkable: series) }
+
+        before do
+          run_all_indexing_jobs
+        end
+
+        it "sorts bookmarkables correctly when logged in" do
+          User.current_user = create(:user)
+          results = BookmarkSearchForm.new(parent: tag, sort_column: "word_count").bookmarkable_search_results
+          # "Series to be bookmarked": 15, "Ten": 10
+          # Check word count of returned results
+          expect(results.map { |item| item.respond_to?(:general_word_count) ? item.general_word_count : item.word_count }).to eq [15, 10]
+          # Check titles of returned results
+          expect(results.map(&:title)).to eq ["Series to be bookmarked", "Ten"]
+        end
+
+        it "sorts bookmarkables correctly when not logged in" do
+          User.current_user = nil
```

Comment:
Is this to fix a failure? If yes, then we have an issue with the inter-test cleanup. Otherwise, I would prefer a `context` block to clarify that this is for logged out users/guests

Reply:
I think I was just confused at the time I wrote this about how inter-test cleanup works. I can remove the line setting current user to nil!